### DISCUSSION TITLE
Change of API data formatting

### DISCUSSION THREAD

**Discussion by theblaz** [qdm12/gluetun#2551] [Q&A]
I'm using Homepage as a dashboard and they have an API functionality to several services including gluetun. I upgraded to the latest version of gluetun and am now getting API errors on homepage. Reached out to homepage and they are asking if gluetun has changed the data formatting. If it has they are willing to modify to accommodate.

Here's the link to the response from the Homepage team 
https://github.com/gethomepage/homepage/discussions/4195#discussioncomment-11058933

I am currently using the apikey as the authentication method and using the Public IP route /v1/publicip/ip

This is the error log in Homepage related to the API

2024-10-28T18:32:21.981Z] error: <validateWidgetData> Invalid data for widget 'gluetun' endpoint 'publicip/ip':
Expected:public_ip,region,country
Parse error: none
Data: {"type":"Buffer","data":[123,34,112,117,98,108,105,99,95,105,112,34,58,34,57,49,46,49,52,56,46,50,52,49,46,52,50,34,44,34,99,111,117,110,116,114,121,34,58,34,84,104,101,32,78,101,116,104,101,114,108,97,110,100,115,34,44,34,108,111,99,97,116,105,111,110,34,58,34,53,50,46,51,56,50,52,48,49,44,52,46,56,57,57,53,48,48,34,44,34,111,114,103,97,110,105,122,97,116,105,111,110,34,58,34,69,119,101,107,97,32,73,110,116,101,114,110,101,116,32,83,101,114,118,105,99,101,115,32,66,46,86,46,34,44,34,116,105,109,101,122,111,110,101,34,58,34,69,117,114,111,112,101,47,65,109,115,116,101,114,100,97,109,34,125,10]}

**Comment by epic0421** [comment]
This is what my config.toml looks like for gluetun.

```
[[roles]]
name = "qbittorrent"
# Define a list of routes with the syntax "Http-Method /path"
routes = ["GET /v1/publicip/ip"]
# Define an authentication method with its parameters
auth = "apikey"
apikey = "*****"
```
Also, in your homepage config, you don't need the fields section if you want all three of them to be visible.

**Reply by theblaz** [reply]
Thanks for getting back to me. I verified that my config.toml file is the same (with the exception of the apikey of course). I also removed the fields section in homepage config. Unfortunately I am still seeing the same error.
![Homepage_error](https://github.com/user-attachments/assets/2b71aa72-ee73-4506-89f0-8be47e44c18c)

**Comment by suprexus** [comment]
I'm having this same issue. Started today when I updated to the latest version. Curl from the homepage container returns:

curl gluetun:8000/v1/publicip/ip
{"public_ip":"91.148.247.195","country":"NL","city":"Lelystad","location":"0.000000,0.000000"}

**Reply by suprexus** [reply]
Went back to 3.38.0. This is the response from gluetun now and homepage can pull the data correctly. 

curl gluetun:8000/v1/publicip/ip
{"public_ip":"81.171.72.11","region":"South Holland","country":"Netherlands","city":"The Hague","location":"52.0767,4.2986","organization":"AS34343 Eweka Internet Services B.V.","postal_code":"2512","timezone":"Europe/Amsterdam"}
